org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/format/LSPFormatter.java

```diff
@@ -45,34 +45,35 @@ public void applyEdits(IDocument document, List<? extends TextEdit> edits) {
 	}
 
 	public CompletableFuture<List<? extends TextEdit>> requestFormatting(@NonNull IDocument document,
-			@NonNull ITextSelection textSelection) {
+			@NonNull ITextSelection textSelection, long expectedModificationStamp) {
 		Collection<@NonNull LSPDocumentInfo> infos = LanguageServiceAccessor.getLSPDocumentInfosFor(document,
 				LSPFormatter::supportFormatting);
 		if (infos.isEmpty()) {
 			return CompletableFuture.completedFuture(Collections.emptyList());
 		}
-		// TODO consider a better strategy for that, maybe iterate on all LS until one gives a result
+		// TODO consider a better strategy for that, maybe iterate on all LS until one
+		// gives a result
 		LSPDocumentInfo info = infos.iterator().next();
+
 		try {
-			return requestFormatting(info, textSelection);
+			return requestFormatting(info, textSelection, expectedModificationStamp);
 		} catch (BadLocationException e) {
 			LanguageServerPlugin.logError(e);
 		}
 		return CompletableFuture.completedFuture(Collections.emptyList());
 	}
 
 	private CompletableFuture<List<? extends TextEdit>> requestFormatting(LSPDocumentInfo info,
-			ITextSelection textSelection) throws BadLocationException {
+			ITextSelection textSelection, long expectedModificationStamp) throws BadLocationException {
 		TextDocumentIdentifier docId = new TextDocumentIdentifier(info.getFileUri().toString());
 		ServerCapabilities capabilities = info.getCapabilites();
 		IPreferenceStore store = EditorsUI.getPreferenceStore();
 		int tabWidth = store.getInt(AbstractDecoratedTextEditorPreferenceConstants.EDITOR_TAB_WIDTH);
 		boolean insertSpaces = store.getBoolean(AbstractDecoratedTextEditorPreferenceConstants.EDITOR_SPACES_FOR_TABS);
-		// use range formatting if standard formatting is not supported or text is selected
-		if (capabilities != null
-				&& isDocumentRangeFormattingSupported(capabilities)
-				&& (!isDocumentFormattingSupported(capabilities)
-						|| textSelection.getLength() != 0)) {
+		// use range formatting if standard formatting is not supported or text is
+		// selected
+		if (capabilities != null && isDocumentRangeFormattingSupported(capabilities)
+				&& (!isDocumentFormattingSupported(capabilities) || textSelection.getLength() != 0)) {
 			DocumentRangeFormattingParams params = new DocumentRangeFormattingParams();
```

Comment:
this is only a change in format, right? If so I would revert it to keep the diff clean

Reply:
Yes, fair enough. I think I might have run auto-format on the file but with subtly different settings.

Is there a recommended `formatter.xml` for the project?